### PR TITLE
add three plugins of xinghuo bif

### PR DESCRIPTION
**Pull request by drafish**
This pr will add three plugins of Xinghuo BIF
1. solidity compiler
2. solidity udapp
3. javascript udapp

Here is the tips videos
1. https://www.bilibili.com/video/BV1Ni4y1h7Yw
2. https://www.bilibili.com/video/BV1uQ4y1g7CH

Here is some links of Xinghuo BIF.
1. office website -- https://bitfactory.cn/english/about-us.html
2. twitter profile -- https://twitter.com/astron_xinghuo
3. office website of CAICT (org behind it) -- http://www.caict.ac.cn/english/about/202004/t20200413_279063.html

Here is a youtube video -- https://www.youtube.com/watch?v=1kC41Zh0V1Y , which is the third seminar of “Blockchain Connecting to the World” International Series Webinars, Co-hosted by ABI, Hyperledger and Cointelegraph China. Xinghuo BIF was one of the attendees. Introduction about Xinghuo BIF starts from 22min to 34min, which may help you learn about Xinghuo BIF.

Here is a documentation to introduce Xinghuo BIF and CAICT.
[CAICT&Intro of Xinghuo BIF-common version_10.10.pdf](https://github.com/ethereum/remix-plugins-directory/files/13910635/CAICT.Intro.of.Xinghuo.BIF-common.version_10.10.pdf)

### CODE REVIEW

**Comment by drafish** [comment]
@ryestew @yann300 Can you check this PR first? We are about to pulish an article for these plugins. We can't do that until this PR is merged.

**Comment by yann300** [comment]
@drafish thanks for the PR and your work!
Could you give us more information about these plugins:
 - `solidity compiler` is this the same as the classic one in Remix? what are the differences?
 - same question for `solidity udapp`.
 - `javascript udapp`: what is the purpose of this one?

Thanks very much!

**Comment by drafish** [comment]
1. `solidity compiler` is almost the same as the classic one in Remix. The only difference between them is the compiler. Xinghuo BIF made some changes based on the original solidity, so the original compiler won't work, we have to use their special compiler.
2. `solidity udapp` is almost the same as the classic one in Remix too.  But the classic one is for Ethereum. We can't deploy contract to Xinghuo BIF in the classic one. That's why this one is here.
3. `js udapp` is for deploying js contract to Xinghuo BIF. JS is another contract language that Xinghuo BIF supports.

@yann300

**Comment by yann300** [comment]
> 1. `solidity compiler` is almost the same as the classic one in Remix. The only difference between them is the compiler. Xinghuo BIF made some changes based on the original solidity, so the original compiler won't work, we have to use their special compiler.
> 2. `solidity udapp` is almost the same as the classic one in Remix too.  But the classic one is for Ethereum. We can't deploy contract to Xinghuo BIF in the classic one. That's why this one is here.
> 3. `js udapp` is for deploying js contract to Xinghuo BIF. JS is another contract language that Xinghuo BIF supports.
> 
> @yann300

Thanks for the answer, could you send me some references where I can learn more about these differences?

**Comment by drafish** [comment]
This is the documentation that explain the difference between xinghuo solidity and ethereum solidity -- https://bif-doc.readthedocs.io/zh-cn/2.0.0/contract/Solidity%E5%90%88%E7%BA%A6%E4%BB%8B%E7%BB%8D.html#soliditysolidity

This is the documentation for the JS contract -- https://bif-doc.readthedocs.io/zh-cn/2.0.0/contract/Javascript%E5%90%88%E7%BA%A6%E4%BB%8B%E7%BB%8D.html

These documentations are chinese, you need to do a quick translation. @yann300

**Comment by drafish** [comment]
Can you guys merge it as soon as possible? @yann300 @ryestew

File: plugins/bif-solidity/profile.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "description": "编译星火链网的 solidity 合约",
```
Comment (yann300) [inline plugins/bif-solidity/profile.json:2]: could you update the description and the display name to be in english.
The description could include something like "Mandarin". thanks

Comment (drafish) [reply plugins/bif-solidity/profile.json]: updated

**Review by yann300 — APPROVED** [review]

**Pull request merged**